backend/src/Designer/Infrastructure/AnsattPorten/AnsattPortenExtensions.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.Studio.Designer.Configuration;
+using Altinn.Studio.Designer.Constants;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace Altinn.Studio.Designer.Infrastructure.AnsattPorten;
+
+public static class AnsattPortenExtensions
+{
+    public static IServiceCollection AddAnsattPortenAuthenticationAndAuthorization(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddAnsattPortenAuthentication(configuration);
+        services.AddAnsattPortenAuthorization(configuration);
+        return services;
+    }
+    private static IServiceCollection AddAnsattPortenAuthentication(this IServiceCollection services, IConfiguration configuration)
+    {
+        bool ansattPortenFeatureFlag = configuration.GetSection($"FeatureManagement:{StudioFeatureFlags.AnsattPorten}").Get<bool>();
+        if (!ansattPortenFeatureFlag)
+        {
+            return services;
+        }
+
+        AnsattPortenLoginSettings oidcSettings = configuration.GetSection(nameof(AnsattPortenLoginSettings)).Get<AnsattPortenLoginSettings>();
+
+        services
+            .AddAuthentication(AnsattPortenConstants.AnsattpotenCookiesAuthenticationScheme)
+            .AddCookie(AnsattPortenConstants.AnsattpotenCookiesAuthenticationScheme, options =>
+            {
+                options.Cookie.HttpOnly = true;
+                options.Cookie.SecurePolicy = CookieSecurePolicy.Always;
+                // options.Cookie.SameSite = SameSiteMode.Lax;
+                options.Cookie.IsEssential = true;
+
+                options.ExpireTimeSpan = TimeSpan.FromMinutes(oidcSettings.CookieExpiryTimeInMinutes);
+                options.SlidingExpiration = true;
+
+                options.Events.OnRedirectToAccessDenied = context =>
+                {
+                    context.Response.StatusCode = StatusCodes.Status403Forbidden;
+                    return Task.CompletedTask;
+                };
+            })
+            .AddOpenIdConnect(AnsattPortenConstants.AnsattportenAuthenticationScheme,
+                options =>
+                {
+                    options.Authority = oidcSettings!.Authority;
```

Comment:
Instead of using "!" (null-forgiving) syntax, what about throwing an ArgumentNullExpection? 
https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=net-8.0

Reply:
We don't have nullable reference types enabled in our project. We're using nullable-oblivious context.
I first implemented Ansattporten integration on my plain new API (with nullable enabled) and later I copied the code, therefore this syntax 😄  I'll just remove null-forgiving syntax since we're not using nullable ref types in project currently.